rpc/rpc.go

```diff
@@ -75,9 +75,10 @@ type RPC struct {
 	running      int32
 }
 
-func New(cfg *config.Config, client *http.Client) (*RPC, error) {
-	if client == nil {
-		client = &http.Client{Timeout: 10 * time.Second}
+func New(cfg *config.Config, transport http.RoundTripper) (*RPC, error) {
+	client := &http.Client{
+		Timeout:   30 * time.Second,
+		Transport: transport,
 	}
```

Comment:
Imho it'd be better if you passed `*http.Client` as before and defined the timeout in some parent function?

Or what is the reason you've switched to passing transport instead?

Reply:
I wanted to have the Timeout configured in a single place as it's cleaner this way. I also thought I'd be able to add a test for it (and I still would like that) but in the current setup we use a mock in place of Stack API. I'd like to move to running a httptest server instead to use the client, this way we'd be able to better test it. But this work will have to wait